core/src/test/java/org/apache/calcite/sql/type/SqlTypeUtilTest.java

```diff
@@ -253,7 +257,17 @@ private void compareTypesIgnoringNullability(
     }
   }
 
+  @Test void testCanCastDifferentConformance() {
```

Comment:
What is this test testing? Add a comment. Don't force future maintainers to read the source code of the test.